components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/APIClient.java

```diff
@@ -202,13 +221,24 @@ private void handleServerError(ActionInvocationResponse.Builder builder, HttpEnt
             builder.errorLog("Unexpected error response received for the status code: " + statusCode + ". "
                     + e.getMessage());
             builder.retry(true);
+            ActionInvocationErrorResponse invalidResponse = handleInvalidResponse();
+            builder.response(invalidResponse);
         }
     }
 
-    private ActionInvocationSuccessResponse handleSuccessResponse(HttpEntity responseEntity)
+    private ActionInvocationErrorResponse handleInvalidResponse() {
+
+        return new ActionInvocationErrorResponse.Builder()
+                .actionStatus(ActionInvocationResponse.Status.ERROR)
+                .error(SERVER_ERROR_CODE)
+                .errorDescription(SEVER_ERROR_MESSAGE)
+                .build();
+    }
```

Comment:
IMO, this method is not needed and all constants here not needed as well

Reply:
This has been utilized to communicate the following exception scenarios from the token endpoint as a `server error (500)`. If that functionality is not mandatory we could remove this.